pkg/workflows/management/install_new_components.go

```diff
@@ -9,21 +9,28 @@ import (
 
 type installNewComponents struct{}
 
-// Run installNewComponents performs actions needed to upgrade the management cluster.
-func (s *installNewComponents) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+func installNewComponentsTask(ctx context.Context, commandContext *task.CommandContext) error {
```

Comment:
For what are you optimizing with this func? What value do we gain in maintainability or understandability by having it instead of putting its logic directly in `func (s *installNewComponents) Run`?

Reply:
Same as above